src/rdata.cpp

```diff
@@ -136,39 +136,36 @@ void ReturnData::applyChainRuleFactorToSimulationResults(const Model *model) {
      * in the sensitivities of simulation results
      * @param model Model from which the ReturnData was obtained
      */
-    if (pscale == AMICI_SCALING_NONE)
-        return;
 
     // chain-rule factor: multiplier for am_p
-    realtype coefficient;
+    std::vector<realtype> coefficient(nplist, 1.0);
 
-    std::vector<realtype> pcoefficient(nplist);
+    std::vector<realtype> pcoefficient(nplist, 1.0);
     std::vector<realtype> unscaledParameters(np);
     model->unscaleParameters(unscaledParameters.data());
     std::vector<realtype> augcoefficient(np);
 
-    switch (pscale) {
+    for (int ip = 0; ip < nplist; ++ip) {
+        switch (pscale[model->plist(ip)]) {
         case AMICI_SCALING_LOG10:
-            coefficient = log(10.0);
-            for (int ip = 0; ip < nplist; ++ip)
-                pcoefficient.at(ip) = unscaledParameters[model->plist(ip)] * log(10);
+            coefficient.at(ip) = log(10.0);
+            pcoefficient.at(ip) = unscaledParameters.at(model->plist(ip)) * log(10);
             if (sensi == AMICI_SENSI_ORDER_SECOND && o2mode == AMICI_O2MODE_FULL)
-                for (int ip = 0; ip < np; ++ip)
-                    augcoefficient.at(ip) = unscaledParameters.at(ip) * log(10);
+                // TODO @REVIEW: unscaledParameters.at(model->plist(ip)) or unscaledParameters.at(ip)
```

Comment:
`unscaledParameters.at(ip)`, here we always have the default order set at compile time (as we need to encode this in the states), which is not affected by the states.
We should actually change this, but for now the previuous implementation was correct.